src/main/java/Duke.java

```diff
@@ -7,6 +7,54 @@
 
 
 public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (FileNotFoundException e) {
+            System.out.println("file not found exception when loading database");
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
+                ui.showLine(); // show the divider line ("_______")
+                Command c = Parser.parse(fullCommand); //throws DukeException
+                c.execute(tasks, ui, storage);
+                isExit = c.isExit();
+            } catch (DukeException e) {
+                ui.showError(e.getErrorMessage());
+            } finally {
+                ui.showLine();
+            }
+        }
+
+        // saving tasks
+        try {
+            ui.showSavingTasks();
+            ui.showLine();
+            storage.save(tasks);
+        } catch (IOException e) {
+            System.out.println("error saving tasks");
+            e.printStackTrace();
+        }
+    }
+
+    public static void main(String[] args) {
+        new Duke("C:\\Users\\Pang Jia Da\\Desktop\\CS2103\\duke\\data\\duke.txt").run();
```

Comment:
Potentially problematic to hard code file locations in your application. Consider using relative locations instead. Additionally, the Windows formatting of file extensions does not play well with other OS.